pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1788,10 +1788,12 @@ def compute_decomposition(wires: WiresLike) -> list[qml.operation.Operator]:
         ]
 
     def pow(self, z: Union[int, float]) -> list[qml.operation.Operator]:
-        z_mod2 = z % 2
-        if abs(z_mod2 - 0.5) < 1e-6:
+        z_mod4 = z % 4
+        if abs(z_mod4 - 0.5) < 1e-6:
             return [SISWAP(wires=self.wires)]
-        return super().pow(z_mod2)
+        if z_mod4 == 2:
+            return [qml.Z(wires=self.wires[0]), qml.Z(wires=self.wires[1])]
+        return super().pow(z_mod4)
```

Comment:
```suggestion
        if abs(z_mod4 - 2) < 1e-6:
            return [qml.Z(wires=self.wires[0]), qml.Z(wires=self.wires[1])]
        return super().pow(z_mod4)
```
BTW I believe we need to take care of `z_mod4==3` as well since the default `pow` method only consider 0 and 1 case lmao